tests/irdl/test_attr_constraint.py

```diff
@@ -56,27 +56,18 @@ class Base(ParametrizedAttribute, ABC):
 class AttrA(Base):
     name = "test.attr_a"
 
-    def __init__(self):
-        super().__init__(())
-
 
 @irdl_attr_definition
 class AttrB(Base):
     name = "test.attr_b"
 
     param: ParameterDef[AttrA]
 
-    def __init__(self, param: AttrA):
-        super().__init__((param,))
-
 
 @irdl_attr_definition
 class AttrC(Base):
     name = "test.attr_c"
 
-    def __init__(self):
-        super().__init__(())
-
```

Comment:
On main there is now an `AttrD` below that needs its init removing

Reply:
Yep but with your change it no longer breaks if it's not removed! I'll remove it anyway.